src/toxcore/state_format/mod.rs

```diff
@@ -31,4 +31,4 @@ https://zetok.github.io/tox-spec/#state-format
 
 
 // FIXME: use new dht code instead of old
-// pub mod old;
+pub mod new;
```

Comment:
Why did you create a new state_format? Is it compatible to old one?

Reply:
Old format has the bytes which need for C language. e.g.: null padding and delimiters for separation of fields.
New format do not have section concepts. I think that section concept is not needed.
New format is not compatible with old format.

Reply:
Why did you do that?

Reply:
https://github.com/tox-rs/tox/blob/3c75733792deca27ef42a7655268467999b5d444/src/toxcore/state_format/old.rs#L624-L635

The padding for fixed size section is not good choice(I think). It just increase code's complexity.

https://github.com/tox-rs/tox/blob/3c75733792deca27ef42a7655268467999b5d444/src/toxcore/state_format/old.rs#L1004-L1044

SectionData has no meaning. It also increases code's complexity.

Reply:
But with this change we won't be able to load profile saved by `c-toxcore`? Or even migrate it? I'd prefer backward compatible option.

Reply:
In c-toxcore, the engine itself produces bytes stream, not a saved file.
And take bytes stream as a parameter to load.
How to save these bytes stream to file(s) or how to load these bytes stream from file(s) is up to the user of c-toxcore.
So, two Tox clients which use same c-toxcore can not guarantee it's compatibility.

Reply:
In the future if we make backward incompatible changes we should consider using something standardized (like [cbor](http://cbor.io/)) and not inventing a new format.

Reply:
Good catch!!!.
I didn't think that our tox-rs may be changed in the future.
And it should provide backward compatibility.

Reply:
> How to save these bytes stream to file(s) or how to load these bytes stream from file(s) is up to the user of c-toxcore.
So, two Tox clients which use same c-toxcore can not guarantee it's compatibility.

But this byte stream is defined by toxcore? I mean how it's possible that different clients save the same stream to a file in a different ways and get different incompatible formats?

Reply:
Ok. I got what you say.
I will modify this PR to use same format, or to provide backward compatibility at least.